scripts/zephyr_module.py

```diff
@@ -370,14 +370,15 @@ def process_sysbuildkconfig(module, meta):
                  'not point to a valid Kconfig file.'
                  .format(module_yml, kconfig_setting))
 
-    if kconfig_setting is None:
-        return ""
+    if kconfig_setting is not None:
+        kconfig_file = os.path.join(module, kconfig_setting)
+        if os.path.isfile(kconfig_file):
+            return kconfig_snippet(meta, module_path, Path(kconfig_file))
 
-    kconfig_file = os.path.join(module, kconfig_setting)
-    if os.path.isfile(kconfig_file):
-        return kconfig_snippet(meta, module_path, Path(kconfig_file))
-    else:
-        return ""
+    name_sanitized = meta['name-sanitized']
+    return (f'config ZEPHYR_{name_sanitized.upper()}_MODULE\n'
```

Comment:
This will only add `ZEPHYR_<module_name>_MODULE` when there is no sysbuild Kconfig file present.
I believe we need to have handling similar to the Zephyr flow here:
https://github.com/zephyrproject-rtos/zephyr/blob/d590bcb5e00b410d5a8631b17d837bc0027d4841/scripts/zephyr_module.py#L320

So that the module is also defined when the there is a Kconfig.

Reply:
commas inside the array creation in `kconfig_snippet()` tricked me.

Code is ok.